core/vm/evm.go

```diff
@@ -456,9 +473,26 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 
 }
 
+// TODO Pass in actual contract ref
+func (evm *EVM) OvmCreate(caller ContractRef, contractAddr common.Address, code []byte, gas uint64, value *big.Int) (ret []byte, retContractAddr common.Address, leftOverGas uint64, err error) {
+	return evm.create(caller, &codeAndHash{code: code}, gas, value, contractAddr)
+}
+
 // Create creates a new contract using code as deployment code.
 func (evm *EVM) Create(caller ContractRef, code []byte, gas uint64, value *big.Int) (ret []byte, contractAddr common.Address, leftOverGas uint64, err error) {
-	contractAddr = crypto.CreateAddress(caller.Address(), evm.StateDB.GetNonce(caller.Address()))
+	// If this is the ExecutionManager & we have already finish initalization, then we should create the contract where it wants
+	isExecutionMgrDoneInitializing := evm.StateDB.GetNonce(ExecutionManagerAddress) > 6 // The EM creates 6 contracts during initalization
```

Comment:
Would make 6 a const somewhere for visibility + self-documentation

Reply:
Got rid of this horrible logic thank goodness lol